sqld/src/auth.rs

```diff
@@ -119,6 +120,40 @@ impl Auth {
     }
 }
 
+impl Authenticated {
+    pub fn from_proxy_grpc<T>(req: &tonic::Request<T>) -> Result<Self, Status> {
+        let auth = match req
+            .metadata()
+            .get(GRPC_PROXY_AUTH_HEADER)
+            .map(|v| v.to_str())
+            .transpose()
+            .map_err(|_| Status::invalid_argument("missing x-proxy-auth"))?
+        {
+            Some("full_access") => Authenticated::Authorized(Authorized::FullAccess),
+            Some("read_only") => Authenticated::Authorized(Authorized::ReadOnly),
+            Some("anonymous") => Authenticated::Anonymous,
+            Some(_) => return Err(Status::permission_denied("invalid authorization level")),
```

Comment:
let's make future debugging easier and print the invalid level as well (`invalid authorization level: {level}`), e.g. when somebody introduces a typo